File: benchmarks/workloads/sum_range_foldable/python/bench.py
import sys

def range_gen(n):
    i = 0
    while i < n:
        yield i
        i += 1

def main():
    n = int(sys.argv[1])
    s = 0
    for v in range_gen(n):
        s = (s + v) & 0xFFFFFFFFFFFFFFFF
    print(f"sum = {s}")

if __name__ == "__main__":
    main()

File: benchmarks/workloads/sum_range_foldable/javascript/bench.js
function* rangeGen(n) {
    for (let i = 0n; i < n; i++) yield i;
}

const MASK = (1n << 64n) - 1n;
const n = BigInt(process.argv[2]);
let s = 0n;
for (const v of rangeGen(n)) {
    s = (s + v) & MASK;
}
console.log(`sum = ${s}`);
